src/common/snippets/src/utils/tokenization_utils.cpp

```diff
@@ -445,53 +446,45 @@ bool tokenize_node(const std::shared_ptr<ov::Node>& node, const SnippetsTokeniza
-                if (constant->get_output_target_inputs(0).size() == 1) {
-                    body_inputs.push_back(input.get_source_output());
-                } else {
+                // If not all Constant consumers are inside Subgraph body,
+                // we should make a copy of this Constant for Subgraph body.
+                if (constant->get_output_target_inputs(0).size() > 1) {
                     const auto constant_consumers = constant->get_output_target_inputs(0);
-                    bool all_consumers_are_inside =
-                        std::all_of(constant_consumers.begin(),
+                    bool has_external_consumers =
+                        std::any_of(constant_consumers.begin(),
                                     constant_consumers.end(),
                                     [&ordered_ops](const ov::Input<ov::Node>& input) {
                                         return std::find(ordered_ops.begin(),
                                                          ordered_ops.end(),
-                                                         input.get_node()->shared_from_this()) != ordered_ops.end();
+                                                         input.get_node()->shared_from_this()) == ordered_ops.end();
                                     });
-                    if (all_consumers_are_inside) {
-                        body_inputs.push_back(input.get_source_output());
-                    } else {
+                    if (has_external_consumers) {
                         const auto constant_copy = constant->clone_with_new_inputs({});
                         node->set_argument(input.get_index(), constant_copy);
-                        body_inputs.emplace_back(constant_copy);
                     }
                 }
             } else if (std::find(ordered_ops.begin(), ordered_ops.end(), parent) == ordered_ops.end()) {
-                auto parameter =
-                    std::make_shared<ov::opset1::Parameter>(input.get_element_type(), input.get_partial_shape());
-                body_parameters.push_back(parameter);
-                body_parameters.back()->set_friendly_name(input.get_node()->get_friendly_name());
-                body_inputs.push_back(parameter->output(0));
-
-                subgraph_inputs.push_back(input.get_source_output());
-
+                const auto& parent_output = input.get_source_output();
+                auto it = std::find(subgraph_inputs.begin(), subgraph_inputs.end(), parent_output);
+                if (it == subgraph_inputs.end()) {
+                    auto new_param =
+                        std::make_shared<ov::op::v0::Parameter>(input.get_element_type(), input.get_partial_shape());
+                    new_param->set_friendly_name(input.get_node()->get_friendly_name());
+                    subgraph_inputs.push_back(parent_output);
+                    body_parameters.push_back(new_param);
+                    it = subgraph_inputs.end() - 1;
```

Comment:
Do we need to assert that size() >= 1 ?

Reply:
We have just pushed `parent_output` to this vector 2 lines before. I think the assertion here is excessive